salt/netapi/rest_cherrypy/app.py

```diff
@@ -1210,29 +1210,21 @@ def GET(self, jid=None, timeout=''):
         '''
         lowstate = [{
             'client': 'runner',
-            'fun': 'jobs.lookup_jid' if jid else 'jobs.list_jobs',
+            'fun': 'jobs.list_job' if jid else 'jobs.list_jobs',
             'jid': jid,
         }]
 
-        if jid:
-            lowstate.append({
-                'client': 'runner',
-                'fun': 'jobs.list_job',
-                'jid': jid,
-            })
-
         cherrypy.request.lowstate = lowstate
         job_ret_info = list(self.exec_lowstate(
             token=cherrypy.session.get('token')))
 
         ret = {}
         if jid:
-            job_ret, job_info = job_ret_info
-            ret['info'] = [job_info]
+            ret['info'] = job_ret_info[0]
+            ret['return'] = [dict((k, job_ret_info[0]['Result'][k]['return']) for k in job_ret_info[0]['Result'])]
```

Comment:
I'm not sure I follow the intent of this list comprehension. Would the following suffice?

```
job_info = job_ret_info[0]
job_ret = job_info.get('Result', {})
[...]
ret['return'] = [job_ret]
```